### PR TITLE
service id changes

### PR DESCRIPTION
**Pull request by sahitya-desireddy**
ServiceID Changes

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/cdapi/service/impl/CrdServiceImplTest.java [hunk 1]
```diff
@@ -60,8 +61,6 @@ void retrieveCategoriesByCategoryId() {
         assertEquals(listOfValueDtos.get(0).getHintTextCy(), actualCategory.getHintTextCy());
         assertEquals(listOfValueDtos.get(0).getHintTextEn(), actualCategory.getHintTextEn());
         assertEquals(listOfValueDtos.get(0).getLovOrder(), actualCategory.getLovOrder());
-        assertEquals(listOfValueDtos.get(0).getServiceId(), actualCategory.getServiceId());
-        assertEquals(listOfValueDtos.get(0).getActive(), actualCategory.getActiveFlag());
         assertNull(actualCategory.getChildNodes());
     }
```
Comment (rajkatla-hmcts) [inline src/test/java/uk/gov/hmcts/reform/cdapi/service/impl/CrdServiceImplTest.java:66]: Add test cases for below scenarios:
1. request call with category id and service id with no child records
2. request call with category id and service id with child records

File: src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java [hunk 2]
```diff
@@ -143,10 +143,10 @@ private ListOfValueDto buildListOfValueDto(String categoryId, String key, String
...
         listOfValueDto1.setValueCy(null);
         listOfValueDto1.setValueEn(value);
-        listOfValueDto1.setServiceId("BBA3");
         CategoryKey categoryKey = new CategoryKey();
         categoryKey.setKey(key);
         categoryKey.setCategoryKey(categoryId);
+        categoryKey.setServiceId("BBA3");
         listOfValueDto1.setCategoryKey(categoryKey);
         return listOfValueDto1;
     }
```
Comment (rajkatla-hmcts) [inline src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java:152]: I don't see integration tests here, but i have seen the existing integration tests and i dont see below scenarios covered. pls add test cases for below scenarios in integration tests:
1. request call with category id and service id with no child records
2. request call with category id and service id with child records

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/test/java/uk/gov/hmcts/reform/cdapi/service/impl/CrdServiceImplTest.java (hunk 1 above)
Comment (sahitya-desireddy) [reply src/test/java/uk/gov/hmcts/reform/cdapi/service/impl/CrdServiceImplTest.java:66]: Thanks @rajkatla-hmcts I have done with unit test cases its working fine

File: src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java (hunk 2 above)
Comment (sahitya-desireddy) [reply src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java:152]: Hi @rajkatla-hmcts Thanks for the above comment 
[src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java] is the PACT Test case .... Instead i have added the same above scenarios in the functional test cases please ... Thank you

**Review by SujeethKumarAri — APPROVED** [review]

File: src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java (hunk 2 above)
Comment (sahitya-desireddy) [reply src/contractTest/java/uk/gov/hmcts/reform/cdapi/provider/CommonDataApiProviderTest.java:152]: Hi @rajkatla-hmcts added integration test cases as well ...Thank you

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java [hunk 3]
```diff
@@ -70,11 +83,26 @@ void shouldThrowStatusCode404ForNullCategoryId()
...
+    void shouldRetrieveCategoriesWithServiceIdWithOutChildNodes()
+        throws JsonProcessingException {
+        final var response = (Categories)
+            commonDataApiClient.retrieveCaseFlagsByServiceId("HearingChannel?serviceId=BBA3"
+                                                                 + "&isChildRequired=N",
+                                                             Categories.class, path
+            );
+        assertNotNull(response);
+        assertEquals(4, response.getListOfCategory().size());
+        responseVerificationWithOutChildNodes(response);
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:96]: pls add a assertion to check how many expected child nodes

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java [hunk 4]
```diff
@@ -48,6 +49,18 @@ void shouldRetrieveCategoriesWithAllParamsWithStatusCode200()
...
+    void shouldRetrieveCategoriesWithServiceIdWithChildNodes()
+        throws JsonProcessingException {
+        final var response = (Categories)
+            commonDataApiClient.retrieveCaseFlagsByServiceId("HearingChannel?serviceId=BBA3"
+                                                                 + "&isChildRequired=Y",
+                                                             Categories.class, path
+            );
+        assertNotNull(response);
+        assertEquals(4, response.getListOfCategory().size());
+        responseVerification(response.getListOfCategory().get(0).getChildNodes().get(0));
```
Comment (rajkatla-hmcts) [inline src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:62]: pls add a assertion to check how many expected child nodes

**Review by rajkatla-hmcts — CHANGES_REQUESTED** [review]

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java (hunk 3 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:97]: Done @rajkatla-hmcts

File: src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java (hunk 4 above)
Comment (sahitya-desireddy) [reply src/integrationTest/java/uk/gov/hmcts/reform/cdapi/RetrieveCategoriesIntegrationTest.java:63]: Done @rajkatla-hmcts

**Review by rajkatla-hmcts — APPROVED** [review]

**Pull request merged**